src/betterproto2/__init__.py

```diff
@@ -1087,7 +1093,10 @@ def _from_dict_init(cls, mapping: Mapping[str, Any] | Any) -> Mapping[str, Any]:
 
                 value_cls = cls._betterproto.cls_by_field[f"{field_name}.value"]
 
-                value = {k: _value_from_dict(v, meta.map_meta[1], value_cls) for k, v in value.items()}
+                value = {
+                    _value_from_dict(k, meta.map_meta[0], value_cls): _value_from_dict(v, meta.map_meta[1], value_cls)
```

Comment:
Using `value_cls` for the type of the key is confusing, since `value_cls` is the class of the value. It works since `_value_from_dict` only needs `value_cls` for messages and enums... which cannot be keys of maps. Using a placeholder value such as `type(None)` should probably be better.
```suggestion
                    _value_from_dict(k, meta.map_meta[0], type(None)): _value_from_dict(v, meta.map_meta[1], value_cls)
```